### ISSUE TITLE
[net10.0] DeveloperBalance ProjectList page fails to load correctly

### ISSUE THREAD

**Issue by SuthiYuvaraj** [dotnet/maui#31152]
### Description

In ProjectList Page , List of projects and newly created projects are not displayed in the project list.

### Steps to Reproduce

1. Install and open the Developer Balance app
2. Click the Hamburger icon and navigate to the project
Actual output: EmptyPage with add button is visible

<img width="1077" height="801" alt="Image" src="https://github.com/user-attachments/assets/d95a6c7c-ad55-4ef1-92f5-0862fff5fa11" />

Expected output: Predefined list should be visible with add button.

<img width="929" height="618" alt="Image" src="https://github.com/user-attachments/assets/fbef3987-8f03-4dcc-be7d-800af7806f73" />

### Version with bug

10.0.0-preview.7

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

macOS, Android, iOS

### Relevant log output

```shell

```

**Comment by ghost** [comment]
This issue has been verified in Visual Studio Code 1.103.0 with MAUI versions 10.0.0.Preview 7, 10.0.0.Preview 1 & 9.0.90. This issue has been reproduced in net10.0 samples and working fine in net9.0 samples on all platforms

**Comment by jfversluis** [comment]
Was the CommunityToolkit library version updated? What version is used for that?

**Comment by SuthiYuvaraj** [comment]
Yes @jfversluis , the CommunityToolkit library version was updated (tested on 11.1.1 and 12.2.0), but the issue still reproduces. The root cause lies in the binding context not being updated properly for the EventToCommand behavior. I have addressed this in this PR(#31158).
